hack/internal/tools/go.mod

```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/kueue/internal/tools
 
-go 1.23
+go 1.23.0
+
+toolchain go1.23.2
```

Comment:
```suggestion
go 1.23.0

```

Couldn't we avoid this toolchain version?

Reply:
+1

Reply:
It was automatically added by `go mod tidy` command.

Reply:
What about other go.mod? Did you run there go mod tidy but it didn't add it?

Reply:
Yes.

Reply:
It is okay to add the toolchain, as some dependencies require this version.

Reply:
Makes sense, I'm ok to keep it here then.